File: app-core/src/main/java/ru/micron/persistence/repository/FilmRepository.java
package ru.micron.persistence.repository;

import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import ru.micron.persistence.model.Film;
import ru.micron.persistence.model.User;

public interface FilmRepository extends JpaRepository<Film, String> {

  List<Film> findAllByOrderByImdbRatingDesc();
  List<Film> findAllByTitle(String title);
  List<Film> findAllByYear(String year);
  List<Film> findAllByYearLessThanEqual(String year);
  List<Film> findAllByYearGreaterThanEqual(String year);

  Page<Film> findFilmsByUsers(User user, Pageable pageable);

  Long countFilmsByUsers(User user);

  boolean existsFilmByUsersAndImdbID(User user, String imdbID);
}

File: app-core/src/main/kotlin/ru/micron/persistence/repository/FilmRepository.kt
package ru.micron.persistence.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.micron.persistence.model.Film
import ru.micron.persistence.model.User

@Repository
interface FilmRepository : JpaRepository<Film, String> {
    fun findAllByOrderByImdbRatingDesc(): List<Film>
    fun findAllByTitle(title: String?): List<Film>
    fun findAllByYear(year: String?): List<Film>
    fun findAllByYearLessThanEqual(year: String): List<Film>
    fun findAllByYearGreaterThanEqual(year: String): List<Film>
    fun findFilmsByUsers(user: User, pageable: Pageable): Page<Film>
    fun existsFilmByUsersAndImdbID(user: User, imdbID: String): Boolean
}